data/categories.txt

```diff
@@ -10626,10 +10626,10 @@ zh-Hans:锁匠
 zh-Hant:鎖匠
 
 craft-painter
-en:Painter
+en:House Painter|decorator
```

Comment:
Please, keep "Painter" here.

Reply:
done!

Reply:
No, I mean "Painter" without "House".

Please, take into account here and in other PRs that category matching works (now) for each token.

So, "House" will produce confusing results.

Reply:
wow, we have **a lot** of phrases in categories.txt, e.g. "Dump Station" matching bus and railway stations, "food waste" (for recycling) matching various cafes and food shops... (I've just checked)

I think its better to fix the root issue than going over the file. Those match phrases are good and legit.
E.g. if I will remove "house" here then it won't help the already bad matching situation (its just one small case of many), but no-one will recall to make it back to the "house painter" after the matching issue is fixed.

@biodranik

Reply:
Of course, there is no silver bullet here and we select some reasonable and appropriate trade-off at this moment.

- "Dump station" - is an example where you can't avoid both of the tokens, this is one phrase
- "House painter" - what info does "house" give here? craft-painter can't work in the office? There are OSM division between house and non-house painter? Why it is needed at all here?
- You can always fix the root issue, I don't mind ..

Reply:
"House Painter" is just because its a translation from types_strings.txt and to avoid confusion with an artist.
So yeap, here we can do with just a "painter".

But its worse with a "mini golf" and especially bad with a "lean-to"..

Reply:
We can't force users to type "only shorter versions without spaces". The matching should be fixed. There is already a boolean flag that user typed a category in the UI. Why that flag can't be used for such cases? If the whole phrase matches, then only this type of object is searched for.